File: Entrada de Dados/AlfaNum/alfanum.jl
function VALIDA_HEXA(VALOR)
  for C in VALOR
    CARACTERE = uppercase(C)
    if (CARACTERE in '0':'9' || 
        CARACTERE in 'A':'F' || 
        CARACTERE in 'a':'f')
      return true
    end
  end
  return false
end

function main()
  println("")
  VALOR_HEX = ""
  SAIR_LOOP = false
  while !SAIR_LOOP
    print("Entre valor hexadecimal de 4 caracteres: ")
    VALOR_HEX = readline()
    VALOR_HEX = strip(VALOR_HEX)
    if length(VALOR_HEX) != 4
      println("Valor inválido, deve ter exatamente 4 caracteres.")
      println()
      continue
    end
    if !VALIDA_HEXA(VALOR_HEX)
      println("Valor inválido, apenas caracteres [0-9], [A-F].")
      println()
      continue
    end
    SAIR_LOOP = true
  end
  VALOR_HEX = uppercase(VALOR_HEX)
  println("Valor fornecido: ", VALOR_HEX)
  println()
end

main()

File: Entrada de Dados/AlfaNum/alfanum.cs
using System;

class Program
{
  static bool VALIDA_HEXA(string VALOR)
  {
    foreach (char c in VALOR)
    {
      char CARACTERE = char.ToUpper(c);
      if (!(char.IsDigit(CARACTERE) || (CARACTERE >= 'A' && CARACTERE <= 'F')))
        return false;
    }
    return true;
  }

  static void Main(string[] args)
  {
    Console.WriteLine();
    string VALOR_HEX = "";
    bool SAIR_LOOP = false;
    do
    {
      Console.Write("Entre valor hexadecimal de 4 caracteres: ");
      VALOR_HEX = Console.ReadLine();
      if (VALOR_HEX.Length != 4)
      {
        Console.WriteLine("Valor invalido, deve ter exatamente 4 caracteres.\n");
      }
      else if (!VALIDA_HEXA(VALOR_HEX))
      {
        Console.WriteLine("Valor invalido, apenas caracteres [0-9] e [A-F].\n");
      }
      else
      {
        SAIR_LOOP = true;
      }
    } while (!SAIR_LOOP);
    VALOR_HEX = VALOR_HEX.ToUpper();
    Console.WriteLine("Valor fornecido: " + VALOR_HEX + "\n");
  }
}
